src/utils/cache.rs

```diff
@@ -19,12 +19,29 @@ impl ServiceCache {
         let _: () = self.set_ex(id, dest, timeout).unwrap();
     }
 
+    pub fn invalidate_pattern(&self, pattern: &String) {
+        let keys: Iter<String> = self.scan_match(pattern).unwrap();
+        let pipeline = &mut pipe();
+        for key in keys {
+            pipeline.del(key);
+        }
+        // I don't know why I have to do this .... 3 f*n hours ...
```

Comment:
Now I know a lot more about database and pooling in rocket ... also how redis macros work and what r2d2 does ...

Reply:
maybe @nlordell can explain me why DeRef doesn't work here ... but might be too Redis/R2D2/Rocket specific

Reply:
See :point_down: